File: 0057-insert-interval/0057-insert-interval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        var merged = new ArrayList<int[]>(Arrays.stream(intervals).filter(it -> it[0] < newInterval[0]).toList());
        merged.add(newInterval);
        merged.addAll(
                Arrays.stream(intervals).filter(it -> it[0] >= newInterval[0]).toList()
        );

        var answer = new ArrayList<int[]>();
        answer.add(merged.get(0));

        for (int i = 1; i < merged.size(); ++i) {
            var it = merged.get(i);
            var last = answer.get(answer.size() - 1);
            if (last[1] < it[0]) {
                answer.add(it);
                continue;
            }
            last[0] = Math.min(last[0], it[0]);
            last[1] = Math.max(last[1], it[1]);
        }
        return answer.toArray(int[][]::new);
    }
}

File: 0057-insert-interval/0057-insert-interval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        intervals.append(newInterval)

        intervals.sort()
        answer = [intervals[0]]

        for it in intervals[1:]:
            if answer[-1][1] < it[0]:
                answer.append(it)
                continue
            answer[-1][0] = min(answer[-1][0], it[0])
            answer[-1][1] = max(answer[-1][1], it[1])

        return answer